posix.lisp

```diff
@@ -107,27 +107,35 @@
 (defun %mmap (path size offset open protection mmap)
   (declare (type fixnum open protection mmap))
   (declare (optimize speed))
-  (let ((fd -1))
+  (let ((fd -1)
+        fresh-fd
+        (error-handler (constantly nil)))
     (etypecase path
+      ((and fixnum unsigned-byte)
+        (setf fd path)
+        ;; If an fd is provided, the burden ought to be on the caller to
+        ;; provide the size as well
+        (check-type size unsigned-byte))
       (string
-       (setf fd (u-open path open))
+       (setf fd (u-open path open)
+             fresh-fd fd
+             error-handler (lambda (e)
+                            (declare (ignore e))
+                            (check-posix (= 0 (u-close fd)))))
```

Comment:
Please double-check my logic here.

I believe the original handler used `(/= fd -1)` as a test to see whether the pathname was null, therefore the fd was generated in the string typecase. Since that's no longer true, I altered the logic so that this handler is used only when the pathname is a string.